### PR TITLE
Queues: array based implementation

### PR DESCRIPTION
**Pull request by antonioaversa**

### CODE REVIEW

File: MoreStructures/Queues/LinkedListQueue.cs [hunk 1]
```diff
@@ -79,7 +79,7 @@ public T Dequeue()
     /// </remarks>
     public void Enqueue(T item)
     {
-        var node = new Node(item, Head);
+        var node = new Node(item, null);
```
Comment (andreaguarino) [inline MoreStructures/Queues/LinkedListQueue.cs:82]: why this change?

**Review by andreaguarino — APPROVED** [review]
LGTM

File: MoreStructures/Queues/LinkedListQueue.cs (hunk 1 above)
Comment (antonioaversa) [reply MoreStructures/Queues/LinkedListQueue.cs:82]: It was a bug found adding the UT `EnqueueAndDequeue_WorkWhenExecutedInBursts`, which is a new unit test at IQueue level and hence shared by both `LinkedListQueue` and `ArrayListQueue`.

The issue comes from a copy paste from LinkedListStack, where `Push` happens on the front of the stack and it makes sense for the new `Node` instance to point to `Head` (the `Node` previously on top of the stack).
